services/core-web/src/components/common/PartySelectField.tsx

```diff
@@ -150,7 +150,7 @@ export const PartySelectField: FC<PartySelectFieldProps> = ({
   };
 
   useEffect(() => {
-    if (searchResults?.party?.length > 0) {
```

Comment:
I know sonar cloud generally prefers the optional chaining here, is this no longer an array?

Reply:
this was a bug that was driving me crazy!
Within this if-block is the bit that puts in the "Add a new party" button. So if searchResults.party == [], you would not have the option to add a new party. It's undefined when the search hasn't completed yet, so the falsey check works.